### ISSUE TITLE
Upgrade to 4.1 blocked by reflections problem

### ISSUE THREAD

**Issue by baburdick** [HornsAndHooves/moribus#2]
Updating the Rails requirement to "~> 4.1.0" causes aggregations to fail when accessing `#options` on a reflection (https://github.com/HornsAndHooves/moribus/blob/master/lib/moribus/macros.rb#L109), e.g:
```
  Moribus Delegations should raise NoMethodError if unknown method received
     Failure/Error: has_aggregated :spec_person_name
     NoMethodError:
       undefined method `options' for #<Hash:0x00000101465d78>
     # ./lib/moribus/macros.rb:109:in `has_aggregated'
     # ./spec/moribus_spec.rb:57:in `<class:SpecCustomerInfo>'
     # ./spec/moribus_spec.rb:53:in `block (2 levels) in <top (required)>'
```
Any insight on how to fix?

**Comment by albertosaurus** [comment]
Is it a variation on this? https://github.com/albertosaurus/power_enum_2/commit/083a04c0ffba4add56c837b2505374643a218e96

**Comment by baburdick** [comment]
It looks like the `belongs_to` method used to return an `ActiveRecord::Reflection::AssociationReflection`, but now it returns a `Hash`. In Rails 4.0.x, it returned something like this:
```
=> #<ActiveRecord::Reflection::AssociationReflection:0x00000101b3f888
 @active_record=
  SpecCustomerInfo(id: integer, spec_customer_id: integer, spec_person_name_id: integer, spec_status_id: integer, spec_type_id: integer, is_current: boolean, lock_version: integer, created_at: datetime, updated_at: datetime, previous_id: integer),
 @collection=false,
 @macro=:belongs_to,
 @name=:spec_person_name,
 @options={},
 @plural_name="spec_person_names",
 @scope=nil>
```
vs. 4.1.x:
```
=> {:spec_customer=>
  #<ActiveRecord::Reflection::AssociationReflection:0x00000101eb1070
   @active_record=
    SpecCustomerInfo(id: integer, spec_customer_id: integer, spec_person_name_id: integer, spec_status_id: integer, spec_type_id: integer, is_current: boolean, lock_version: integer, created_at: datetime, updated_at: datetime, previous_id: integer),
   @automatic_inverse_of=nil,
   @collection=false,
   @constructable=true,
   @foreign_key="spec_customer_id",
   @foreign_type="spec_customer_type",
   @klass=nil,
   @macro=:belongs_to,
   @name=:spec_customer,
   @options={:inverse_of=>:spec_customer_info, :touch=>true},
   @plural_name="spec_customers",
   @scope=nil,
   @type=nil>,
 :spec_person_name=>
  #<ActiveRecord::Reflection::AssociationReflection:0x00000101ea8718
   @active_record=
    SpecCustomerInfo(id: integer, spec_customer_id: integer, spec_person_name_id: integer, spec_status_id: integer, spec_type_id: integer, is_current: boolean, lock_version: integer, created_at: datetime, updated_at: datetime, previous_id: integer),
   @automatic_inverse_of=nil,
   @collection=false,
   @constructable=true,
   @foreign_type="spec_person_name_type",
   @klass=nil,
   @macro=:belongs_to,
   @name=:spec_person_name,
   @options={},
   @plural_name="spec_person_names",
   @scope=nil,
   @type=nil>}
```

**Comment by baburdick** [comment]
Adding this line at 108, seems to have allowed me to fail through to the next error:
```ruby
      reflection = reflection[name] if reflection.respond_to? :[]
```
Investigating further ...

**Comment by baburdick** [comment]
Discussion can move to the pull request: HornsAndHooves/moribus#3

**Comment by baburdick** [comment]
Pretty sure this was fixed by HornsAndHooves/moribus#3 .
